### ISSUE TITLE
Past 7 days actually filters for past 8 days

### ISSUE THREAD

**Issue by gitram** [metabase/metabase#3731]
When I construct a query to return all payments in the last 7 days (issued at 2016-11-06 20:52:10.309 UTC) this is the SQL of the query built:

```sql
SELECT count(*) AS [count]
FROM [analytics.payments]
WHERE CAST(date([analytics.payments.createdAt]) AS timestamp) 
      BETWEEN CAST(date(timestamp('2016-10-29T20:52:10.309Z')) AS timestamp)
          AND CAST(date(timestamp('2016-11-05T20:52:10.309Z')) AS timestamp)
```

The problem here is that this includes all transactions for the 2016-10-29, 2016-10-30, 2016-10-31, 2016-11-01, 2016-11-02, 2016-11-03, 2016-11-04, 2016-11-05 which is 8 days since the BETWEEN operator is inclusive. I think the bug in the logic is from the cast(date(.)) operator which rewinds the first date in the BETWEEN clause to the (00:00:00) and second date in the BETWEEN clause to (23:59).

The alternative SQL I use for Bigquery is

```sql
 BETWEEN DATE_ADD(CURRENT_TIMESTAMP(),-7, 'DAY') AND CURRENT_TIMESTAMP()
```

**Comment by moumny** [comment]
Actually this is not specific to BQ. Except for past 1 day( yesterday) past X days return past X+1 day (not including current day). Behavior is the same for Druid and Sample dataset.

**Comment by camsaul** [comment]
This was fixed in version 22.0, @moumny please update to the latest version of Metabase

**Comment by camsaul** [comment]
Fixed by #4104
